### ISSUE TITLE
Allow to override Ajv instance

### ISSUE THREAD

**Issue by leppaott** [seriousme/fastify-openapi-glue#339]
Hey,

Could it be possible to add another option or extend the ajvOptions to use user given ajv instance instead of creating your own. The problem is here: https://github.com/seriousme/fastify-openapi-glue/blob/master/index.js#L60 the usage setValidatorCompiler overrides anything user might want and setValidatorCompiler was pretty much added so that user can override ajv instance on the fastify, but using this tool that's impossible.

I did a small patch https://github.com/seriousme/fastify-openapi-glue/blob/master/index.js#L56 on that line with following:

```
-  const ajv = new Ajv(ajvOptions);
+  const ajv = ajvOpts || new Ajv(ajvOptions);
```

To let me able to pass the ajv instance through the ajvOptions.

Would it be possible to add this as ajvInstance option (override ajv instance which doesn't take ajvOptions into account or such).

**Comment by seriousme** [comment]
Hi,

thanks for asking.

I added the creation of the AJV instance in the early days of fastify-openapi-glue as AJV did not support the required formats.
These days [ajv-formats](https://github.com/ajv-validator/ajv-formats) contains the required formats. So the private AJV instance is no longer really required when using AJV 8. (AJV 8 will be the default for Fastify 4)
I therefore propose to add a boolean option `defaultAJV` that , when set to `true` skips the `setValidatorCompiler()`.
This enables you to do with AJV whatever you like as the plugin will no longer touch the AJV configuration.
It will also mean that you need to add `ajv-formats` yourself to the AJV instance.

What do you think ?

Kind regards,
Hans

**Comment by seriousme** [comment]
Hi, 

Just released version 2.7.0 on NPM with the `defaultAJV` [option](https://github.com/seriousme/fastify-openapi-glue/blob/master/README.md#options).
Here is an example of its usage: 
https://github.com/seriousme/fastify-openapi-glue/blob/b493ff315c6dd263d5faf010253061a9c42cdec5/test/test-plugin.v3.js#L363-L396

Kind regards,
Hans

**Comment by leppaott** [comment]
NPM 2.7.0 still has the old code?

```
   "version": "2.7.0",
...
   "fastify-cli": "^3.1.0",
...
async function fastifyOpenapiGlue(instance, opts) {
  setValidatorCompiler(instance, opts.ajvOptions, opts.noAdditional);
```

**Comment by seriousme** [comment]
Hi,

thanks for spotting the error !  This should be fixed in 2.7.1.

The extra dependency will no longer be required in Fastify 4.0. In the next major version of Fastify-openapi-glue I intend to remove all AJV related configuration as there is no need to have a private instance of AJV anymore in the plugin. 

Kind regards,
Hans

**Comment by leppaott** [comment]
Thanks for the quick help. This helped tremendously us! I shall close this now as this does work.
